exonum/src/api/node/public/explorer.rs

```diff
@@ -59,29 +57,42 @@ pub struct BlockInfo {
     /// Block header as recorded in the blockchain.
     pub block: Block,
     /// Precommits authorizing the block.
-    pub precommits: Vec<Signed<Precommit>>,
+    pub precommits: Option<Vec<Signed<Precommit>>>,
     /// Hashes of transactions in the block.
-    pub txs: Vec<Hash>,
+    pub txs: Option<Vec<Hash>>,
     /// Median time from the block precommits.
-    pub time: DateTime<Utc>,
+    pub time: Option<DateTime<Utc>>,
 }
 
 /// Blocks in range parameters.
 #[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Default)]
 pub struct BlocksQuery {
     /// The number of blocks to return. Should not be greater than `MAX_BLOCKS_PER_REQUEST`.
     pub count: usize,
-    /// The maximum height of the returned blocks. The blocks are returned in reverse order,
-    /// starting from the latest and at least up to the `latest` - `count` + 1.
+    /// The maximum height of the returned blocks.
+    ///
+    /// The blocks are returned in reverse order,
+    /// starting from the latest and at least up to the `latest - count + 1`.
     /// The default value is the height of the latest block in the blockchain.
     pub latest: Option<Height>,
+    /// The minimum height of the returned blocks. The default value is `Height(0)` (the genesis
+    /// block).
+    ///
+    /// Note that `earliest` has the least priority compared to `latest` and `count`;
+    /// it can only truncate the list of otherwise returned blocks if some of them have a lesser
+    /// height.
+    pub earliest: Option<Height>,
     /// If true, then only non-empty blocks are returned. The default value is false.
     #[serde(default)]
     pub skip_empty_blocks: bool,
-    /// If true, then `BlocksRange`'s `times` field will contain median time from the
+    /// If true, then the returned `BlocksRange`'s `times` field will contain median time from the
     /// corresponding blocks precommits.
     #[serde(default)]
     pub add_blocks_time: bool,
+    /// If true, then the returned `BlocksRange.precommits` will contain precommits for the
+    /// corresponding returned blocks.
+    #[serde(default)]
+    pub add_precommits: bool,
```

Comment:
Maybe more clear name is `with_precommits` and `with_blocks_time`?

Reply:
I've used the same naming scheme as was present with `add_blocks_time`, but I'm not against changing it.